### PR TITLE
test: add serialization tests

### PR DESCRIPTION
**Pull request by nechaido**
Based on https://github.com/metarhia/jstp/pull/236.

### CODE REVIEW

File: test/fixtures/json5/arrays/no-comma-array.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
 [
     true
     false
-]
\ No newline at end of file
+]
```
Comment (belochub) [inline test/fixtures/json5/arrays/no-comma-array.txt]: What was changed on this line?

**Comment by nechaido** [comment]
I will be easier to review using [comparison](https://github.com/metarhia/jstp/compare/test-refactor-serde-tests...add-serialization-tests) with #236

File: test/fixtures/json5/serialization/index.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+'use strict';
```
Comment (aqrln) [inline test/fixtures/json5/serialization/index.js]: Why is this in `test/fixtures/json5`?

Comment (nechaido) [inline test/fixtures/json5/serialization/index.js]: I placed it here because they are tests for json5 specification, byt i guess I should have placed them in `test/todo/json5/`

File: test/fixtures/todo/json5/serialization/object.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    value: { 42: true },
+    serialized: '{42:true}'
+  },
+  {
+    name: 'object with unquoted keys',
+    value: { $: 'dollar', _$_: 'multiple symbols' },
+    serialized: '{$:\'dollar\',_$_:\'multiple symbols\'}'
+  },
+  {
+    name: 'object with unicode unquoted key',
```
Comment (belochub) [inline test/fixtures/todo/json5/serialization/object.js]: "unicode" should be capitalized here.

Comment (belochub) [inline test/fixtures/todo/json5/serialization/object.js]: Also, it is better to name this test case "object with identifier name, containing non-latin Unicode literals, as a key".

File: test/fixtures/todo/json5/serialization/object.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
+'use strict';
+
+module.exports = [
+  {
+    name: 'object with numeric literal as key',
+    value: { 42: true },
+    serialized: '{42:true}'
+  },
+  {
+    name: 'object with unquoted keys',
```
Comment (belochub) [inline test/fixtures/todo/json5/serialization/object.js]: And this one should be called "object with identifier names as keys".

File: test/fixtures/serde-test-cases/serialization/string.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
+'use strict';
+
+module.exports = [
+  {
+    name: 'string with inline comment like substring',
```
Comment (belochub) [inline test/fixtures/serde-test-cases/serialization/string.js]: This name is a bit misleading here.

File: test/fixtures/todo/json5/serialization/object.js (hunk 4 above)
Comment (nechaido) [inline test/fixtures/todo/json5/serialization/object.js]: This name was taken from corresponding test for JSON5 parsing test: `test/fixtures/json5/objects/unquoted-keys.json5`.

Comment (belochub) [inline test/fixtures/todo/json5/serialization/object.js]: @nechaido, yes, and these JSON5 tests are imported while you are creating new tests, in my opinion, it is better to name them correctly.

Comment (nechaido) [inline test/fixtures/todo/json5/serialization/object.js]: @belochub ok

File: test/fixtures/serde-test-cases/serialization/string.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
+'use strict';
+
+module.exports = [
+  {
+    name: 'string with substring like inline comment',
```
Comment (belochub) [inline test/fixtures/serde-test-cases/serialization/string.js]: This should be called "string containing inline comment".

File: test/fixtures/todo/json5/serialization/object.js [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+    serialized: '{42:true}'
+  },
+  {
+    name: 'object with identifier names as keys',
+    value: { $: 'dollar', _$_: 'multiple symbols' },
+    serialized: '{$:\'dollar\',_$_:\'multiple symbols\'}'
+  },
+  {
+    name: 'object with identifier name' +
+      ', containing non-latin Unicode literals, as a key',
```
Comment (belochub) [inline test/fixtures/todo/json5/serialization/object.js]: You shouldn't start a new line with a comma.

**Comment by nechaido** [comment]
@aqrln ping.

**Comment by aqrln** [comment]
Landed in 38074469a875761db13335fa395f8383a1e9c055.
